tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -1476,76 +1474,78 @@ public static void test_prepare_insert() throws Exception {
 
 	public static void test_read_only() throws Exception {
 		Path database_file = Files.createTempFile("duckdb-jdbc-test-", ".duckdb");
-		database_file.toFile().delete();
+		Files.deleteIfExists(database_file);
 
-		String jdbc_url = "jdbc:duckdb:" + database_file.toString();
+		String jdbc_url = "jdbc:duckdb:" + database_file;
 		Properties ro_prop = new Properties();
 		ro_prop.setProperty("duckdb.read_only", "true");
 
 		Connection conn_rw = DriverManager.getConnection(jdbc_url);
 		assertFalse(conn_rw.isReadOnly());
+		assertFalse(conn_rw.getMetaData().isReadOnly());
 		Statement stmt = conn_rw.createStatement();
 		stmt.execute("CREATE TABLE test (i INTEGER)");
 		stmt.execute("INSERT INTO test VALUES (42)");
 		stmt.close();
-		// we cannot create other connections, be it read-write or read-only right now
-		try {
-			Connection conn2 = DriverManager.getConnection(jdbc_url);
-			fail();
-		} catch (Exception e) {
```

Comment:
This block of code was actually what was causing this test to fail on Windows. The `conn2` opens successfully. However, we then hit the `fail()`, which throws, but is immediately caught by the `catch`, hiding the problem.

Since the connection opened, there's an additional handle to the database file open. Using `lsof` on Mac, I was able to see that prior to the changes in this PR, Java kept the DB file open after this block, and after the changes, it doesn't.